tests/unit/splinter_test.c

```diff
@@ -138,11 +138,7 @@ CTEST_SETUP(splinter)
    data->splinter_cfg = TYPED_ARRAY_MALLOC(data->hid, data->splinter_cfg,
                                           num_tables);
 
-   data->data_cfg = TYPED_ARRAY_MALLOC(data->hid, data->data_cfg,
-                                       num_tables);
-
-   data->cache_cfg = TYPED_ARRAY_MALLOC(data->hid, data->cache_cfg,
-                                        num_tables);
+   data->cache_cfg = TYPED_ARRAY_MALLOC(data->hid, data->cache_cfg, num_tables);
```

Comment:
What's going on with this `data_cfg` allocation going away?

Reply:
It's actually a funny business. For now, it's a patch-fix, but there is potentially a latent bug waiting to happen, if ever these tests are run with num-tables > 1.

The issue is that if we allocated memory here, we should free it at the end. Otherwise, there is a small memory leak and (I think) ASAN/ MSAN runs will trip up.

But, if you try to call `platform_free()` at the end of the test case on `data->data_cfg`, you will get an error.

That's because we inside this parsing function, we have this code:

```
206 static inline void
207 test_config_init(trunk_config           *splinter_cfg,  // OUT
208                  data_config           **data_cfg,      // OUT
[...]
217    *data_cfg             = test_data_config;
```

Although the code is meaning to say "structure assignment", I think all that is happening here is that `data_cfg` output param is set to the global `test_data_config`'s address.

So, upon return from `test_parse_args_n`, we would have trashed the allocated handle to `data_cfg` ptr.
And an attempt to `platform_free()` at the end will fail with some error; I forget what.

The latent bug is coming from here in `test_parse_args_n`:

```
310    for (i = 0; i < num_config; i++) {
311       test_config_init(&splinter_cfg[i],
312                        &data_cfg[i],
...
```

Here on L310, `num_config` is really supposed to be `num_tables`. 

This area is soft & buggy ... and I did not want to chase all of them down and bloat this change set.

I don't think we are running with num_tables > 1.  The right thing to do would be to exercise such tests, confirm that the behvaiour is buggy and fix the code the right way. But that's ... way out of scope.


??? Makes sense ??

Reply:
I believe the data_cfg thing is a change I made.

I modified the data_config system so that you can sub-class a data_config in the same way that we sub-class other structs:

struct my_data_config {
  data_config super;
  <additional fields go here>;
};

This means that we must never copy data_configs.  We just take the pointer given to us by the application and use that.  That also means we don't allocate memory for data_configs.

Reply:
But, then @rtjohnso - Do you agree with my suspicion that ... "but there is potentially a latent bug waiting to happen, if ever these tests are run with num-tables > 1."?

We can walk-thru code tomorrow when in the office.

I think what I have now in this change-set is adequately correct for the set of tests being run now.

Do you think there is any further need to revise the code as submitted for review?